astroid/rebuilder.py

```diff
@@ -675,13 +671,27 @@ def visit_annassign(self, node: "ast.AnnAssign", parent: NodeNG) -> nodes.AnnAss
         )
         return newnode
 
+    @overload
     def visit_assignname(
-        self, node: "ast.AST", parent: NodeNG, node_name: Optional[str] = None
+        self, node: "ast.AST", parent: NodeNG, node_name: str
     ) -> nodes.AssignName:
+        ...
+
+    @overload
+    def visit_assignname(
+        self, node: "ast.AST", parent: NodeNG, node_name: None
+    ) -> None:
+        ...
+
+    def visit_assignname(
```

Comment:
For me a visit_x function is accessed with:
```python
    def accept(self, visitor):
        """Visit this node using the given visitor."""
        func = getattr(visitor, "visit_" + self.__class__.__name__.lower())
        return func(self)
```
And if it's named ``visit_assignname`` the node is going to be an ``AssignName`` not a NodeNG ? What am I missing ?

Reply:
Actually `accept` isn't ever used for the `rebuilder`, as I can see it, it's just for `as_string`. The rebuilder uses `def visit` instead. However, `visit_assignname` is also never called by `visit` as there is no corresponding `ast` node. It is more a helper function, probably with a not ideal name. If already added a note to the docstring to explain that it's never called by `visit`. (Same is true for `visit_decorators` and `visit_module`)

Reply:
But ``Rebuilder`` look like a ``Visitor`` though, maybe we could refactor it so it's used like one ?

Reply:
OK then.